build-coordinator/src/main/java/org/jboss/pnc/core/builder/coordinator/BuildCoordinator.java

```diff
@@ -217,7 +217,6 @@ void processBuildTask(BuildTask buildTask) {
                         datastoreAdapter.storeResult(buildTask, buildResult);
                         coordinationStatus = BuildCoordinationStatus.DONE_WITH_ERRORS;
                     } else {
-                        removeSubmittedTask(buildTask);
                         throw new BuildCoordinationException("Failed task should have set exception or failed reason status.");
```

Comment:
This can be optimized further, It does not make sense to throw an exception and catch it few lines bellow.